docs/decisions/0002-initial-api-specification.rst

```diff
@@ -0,0 +1,436 @@
+       'user_id': 54321,
+       'content_key': 'demox_1234+2T2023',
+       'quantity': 19900,
+       'unit': 'USD_CENTS',
+       'reference_id': 1234,
+       'reference_table': 'enrollments',
+       'subsidy_access_policy_uuid': 'a-policy-uuid',
+       'metadata': {...},
+       'created': 'created-datetime',
+       'modified': 'modified-datetime',
+       'reversal': null
+   }
+
+Permissions
+-----------
+
+enterprise_learner
+  The transaction object should only be returned if requesting user's ``lms_user_id`` matches the ``user_id``
+  for the transaction record.
+
+enterprise_admin
+  Should return the requested subsidy transaction only if the requesting user has implicit (JWT) or explicit (DB-defined)
+  ``enterprise_admin`` role assigned for the requested subsidy's enterprise.
+
+openedx_operator
+  If the requesting user is implicitly or explicitly granted this role, they have permissions to view **all**
+  subsidy records, and can therefore retrieve the requested subsidy transaction.
+
+POST enterprise subsidy transaction
+===================================
+**/api/v1/subsidies/[subsidy-uuid]/transactions/**
+
+Create a new subsidy transaction in the subsidy's ledger.
+Only service users (those with ``openedx_operator`` role assignment) can do this.
+A side-effect of a successful POST request here is the creation of a course enrollment or entitlement
+that "fulfills" the ledger transaction.
+The subsidy-service should determine the price of the content before creating the transaction, relying
+on the course-discovery service for this data.
+
+Inputs
+------
+
+- ``subsidy-uuid`` (URL path, required): The uuid (primary key) of the subsidy for which transactions should be listed.
+- ``user_id`` (POST data, required): The user for whom the transaction is written and for which a fulfillment should occur.
+- ``content_key`` (POST data, required): The content for which a fulfillment is created.
+- ``access_policy_uuid`` (POST data, required): The uuid of the policy that allowed the ledger transaction to be created.
+
+Outputs
+-------
+Returns data about the transaction.
+
+::
+
+   {
+       'uuid': 'the-transaction-uuid',
+       'status': 'completed',
+       'idempotency_key': 'the-idempotency-key',
+       'user_id': 54321,
+       'content_key': 'demox_1234+2T2023',
+       'quantity': 19900,
+       'unit': 'USD_CENTS',
+       'reference_id': 1234,
+       'reference_table': 'enrollments',
+       'subsidy_access_policy_uuid': 'a-policy-uuid',
+       'metadata': {...},
+       'created': 'created-datetime',
+       'modified': 'modified-datetime',
+       'reversal': null
+   }
+
+Permissions
+-----------
+
+enterprise_learner
+  NOT ALLOWED
+
+enterprise_admin
+  NOT ALLOWED
+
+openedx_operator
+  If the requesting user is implicitly or explicitly granted this role, they have permissions to create
+  transactions for any active ledgered-subsidy.
+
+POST enterprise subsidy transaction reversal
```

Comment:
I get why it's like this but for the record I like it semantically better if this was a delete on the URL previous to this but I get why you specified it this way.